modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -165,6 +166,35 @@ public static string NameWithTypeParameters(this INamedTypeSymbol symbol)
         public static string FullQualifiedName(this INamespaceSymbol namespaceSymbol)
             => namespaceSymbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
 
+        public static string FullQualifiedName(this ISymbol symbol)
+            => symbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
+
+        public static string FullQualifiedSyntax(this SyntaxNode node, SemanticModel sm, bool isFirstNode = true)
+        {
+            if (node is NameSyntax ns && isFirstNode)
+            {
+                SymbolInfo nameInfo = sm.GetSymbolInfo(ns);
+                return nameInfo.Symbol?.FullQualifiedName() ?? ns.ToString();
+            }
+
+            StringBuilder sb = new StringBuilder(128);
+            bool innerIsFirstNode = true;
+            foreach (var child in node.ChildNodesAndTokens())
+            {
+                if (child.HasLeadingTrivia) { sb.Append(child.GetLeadingTrivia()); }
+
+                if (child.IsNode)
+                {
+                    sb.Append(child.AsNode().FullQualifiedSyntax(sm, isFirstNode: innerIsFirstNode));
+                    innerIsFirstNode = false;
+                }
+                else { sb.Append(child); }
+
+                if (child.HasTrailingTrivia) { sb.Append(child.GetTrailingTrivia()); }
+            }
+            return sb.ToString().TrimEnd();
```

Comment:
Is trimming the end here really necessary?

Reply:
No, it used to be when we were adding the spaces, but it doesn't make sense anymore. Removed.